cdi/src/test/java/io/smallrye/config/inject/ValidateInjectionTest.java

```diff
@@ -26,68 +28,212 @@
-                message("SRCFG02000: No Config Value exists for required property missing.property")));
+    void emptyProperty() throws Exception {
+        DeploymentException exception = getDeploymentException(EmptyPropertyTest.class);
+        assertThat(exception)
+                .hasMessageStartingWith("SRCFG02001: Failed to Inject @ConfigProperty for key empty.property SRCFG00040:");
+
+        assertThat(exception.getCause()).isInstanceOf(ConfigInjectionException.class);
+
+        assertThat(exception.getCause().getCause()).isInstanceOf(NoSuchElementException.class);
+        assertThat(exception.getCause().getCause()).hasMessage(
+                "SRCFG00040: The config property empty.property is defined as the empty String (\"\") which the following Converter considered to be null: io.smallrye.config.Converters$BuiltInConverter");
+    }
+
+    @Test
+    void badProperty() throws Exception {
+        DeploymentException exception = getDeploymentException(BadPropertyTest.class);
+        assertThat(exception)
+                .hasMessageStartingWith("SRCFG02001: Failed to Inject @ConfigProperty for key bad.property SRCFG00041:");
+
+        assertThat(exception.getCause()).isInstanceOf(ConfigInjectionException.class);
+
+        assertThat(exception.getCause().getCause()).isInstanceOf(NoSuchElementException.class);
+        assertThat(exception.getCause().getCause()).hasMessage(
+                "SRCFG00041: The config property bad.property with the config value \",\" was converted to null from the following Converter: io.smallrye.config.Converters$ArrayConverter");
+    }
+
+    @Test
+    void customConverterMissingProperty() {
+        DeploymentException exception = getDeploymentException(CustomConverterMissingPropertyTest.class);
+        assertThat(exception).hasMessage(
+                "SRCFG02000: Failed to Inject @ConfigProperty for key missing.property since the config property could not be found in any config source");
+
+        assertThat(exception.getCause()).isInstanceOf(ConfigInjectionException.class);
+        assertThat(exception.getCause()).hasMessage(
+                "SRCFG02000: Failed to Inject @ConfigProperty for key missing.property since the config property could not be found in any config source");
+    }
+
+    @Test
+    void MissingConverter() {
+        DeploymentException exception = getDeploymentException(MissingConverterTest.class);
+        assertThat(exception).hasMessageStartingWith(
+                "SRCFG02001: Failed to Inject @ConfigProperty for key my.prop SRCFG02006:");
+
+        assertThat(exception.getCause()).isInstanceOf(ConfigInjectionException.class);
+
+        assertThat(exception.getCause().getCause()).isInstanceOf(java.lang.IllegalArgumentException.class);
+        assertThat(exception.getCause().getCause()).hasMessage(
+                "SRCFG02006: No Converter registered for class io.smallrye.config.inject.ValidateInjectionTest$MissingConverterTest$MyType");
```

Comment:
This message is not a ConfigMessages (SRCFG0000X) but it could be with a fair bit of re-shuffling of the smallrye-config-core code. Although I doubt think it would be worth it...